### PR TITLE
OCPBUGS-3988: haproxy - use curl for validation

### PR DESCRIPTION
**Pull request by dougsland**
Instead of using `oc get nodes` to make sure the deploy was correct. Use the same approach as others components.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

### CODE REVIEW

File: templates/master/00-master/on-prem/files/haproxy.yaml
Comment (cybertron) [inline templates/master/00-master/on-prem/files/haproxy.yaml:38]: This doesn't do what we need. The point is to verify that we can get to the API via the api-int DNS record, so using localhost won't help. The address needs to be api-int.[cluster domain].

Comment (dougsland) [reply templates/master/00-master/on-prem/files/haproxy.yaml]: thanks Ben.

**Comment by dougsland** [comment]
/unhold

*VERIFIED*
```
$ oc get pods -n openshift-kni-infra
NAME                                            READY   STATUS    RESTARTS   AGE
coredns-master-0.ostest.test.metalkube.org      2/2     Running   0          72m
coredns-master-1.ostest.test.metalkube.org      2/2     Running   0          72m
coredns-master-2.ostest.test.metalkube.org      2/2     Running   0          72m
haproxy-master-0.ostest.test.metalkube.org      2/2     Running   0          72m
haproxy-master-1.ostest.test.metalkube.org      2/2     Running   0          73m
haproxy-master-2.ostest.test.metalkube.org      2/2     Running   0          71m
keepalived-master-0.ostest.test.metalkube.org   2/2     Running   0          71m
keepalived-master-1.ostest.test.metalkube.org   2/2     Running   0          73m
keepalived-master-2.ostest.test.metalkube.org   2/2     Running   0          72m
```

**Comment by cybertron** [comment]
/test e2e-metal-ipi
/test e2e-metal-ipi-ovn-ipv6
/test e2e-vsphere-upgrade

This lgtm, pending ci passes from the on-prem jobs.

**Comment by cybertron** [comment]
/lgtm

I don't have approval rights anyway so I guess there's no need to wait on the lgtm tag.

**Comment by dougsland** [comment]
the timeout errors are not related to the patch

**Review by sinnykumari — APPROVED** [review]
LGTM, on-prem ci tests are passing

**Pull request merged**
